Commit to week-2

AmarKaramehmedovic opened the pull request.

AmarKaramehmedovic committed:
Commit to week-2

antonioorct commented on src/components/SimpleSearch.tsx:
```diff
@@ -0,0 +1,35 @@
+import { MDBBtn } from 'mdb-react-ui-kit';
+
+
+const SimpleSearch = () => {
```
You need to add state to the form and then have an `onSubmit` handler that logs the form's values

antonioorct commented on src/components/AdvancedSearch.tsx:
```diff
@@ -0,0 +1,103 @@
+    accType: Object
+}
+
+const AdvancedSearch = () => {
```
Add state to form and `onSubmit` handler

antonioorct replied:
Be mindful when adding state to the form to keep the form state as an object instead of creating a lot of `useState` calls for each of the form's fields.
Refer to this link for help and an example. Let me know if you get stuck.
https://www.w3schools.com/react/react_forms.asp#:~:text=Multiple%20Input%20Fields

antonioorct commented on src/components/AdvancedSearch.tsx:
```diff
@@ -0,0 +1,103 @@
+
+interface State {
+    location: string;
+    date1: Date;
+    date2: Date;
```
Give these more meaningful names: e.g. `checkInDate`, `checkOutDate`

antonioorct commented on src/components/AdvancedSearch.tsx:
```diff
@@ -0,0 +1,103 @@
+            <div>
+                <TextField
+                    label="How many people?"
+                    id="outlined-start-adornment"
```
You could add `type=number` here to force the input to allow numbers only

antonioorct commented on src/components/AdvancedSearch.tsx:
```diff
@@ -0,0 +1,103 @@
+                        startAdornment: <InputAdornment position="start"><Hotel /></InputAdornment>,
+                    }}
+                >
+                    {accType.map((option) => (
+                        <MenuItem key={option.value} value={option.value}>
+                            {option.label}
+                        </MenuItem>
+                    ))}
```
Good job creating an array of options and then iterating over them with .map

antonioorct commented on src/components/AccomodationSearch.css:
```diff
@@ -0,0 +1,28 @@
+}
+
+.search-btn {
+    width: 148px;
+    height: 56px;
```
You shouldn't have fixed size for buttons. Just define a padding instead

antonioorct commented on src/components/ReservationForm.tsx:
```diff
@@ -0,0 +1,85 @@
+import { Col, Container, Row } from 'react-bootstrap';
+
+
+const ReservationForm = () => {
+
+    const [date1, setDate1] = React.useState<Date | null>(new Date());
+    const [date2, setDate2] = React.useState<Date | null>(new Date());
+
+    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        console.log("Fetch updated filter");
+    };
```
Add state to the form, keep the state in an object and rename the variables to more meaningful names.
Add `onSubmit` handler to print the form values

antonioorct replied:
https://medium.com/@sandupa.egodage/react-form-with-typescript-c74510b2f9d3
Here's an article that regards this

antonioorct reviewed (changes requested):
Make changes regarding the comments. If you need help adding state to the forms let me know and we can go through it together

AmarKaramehmedovic committed:
Fixes on requested changes #1

Fixed everything but couldn't managed to add State without getting an error on "onChange" on non date fields:
Error: The expected type comes from property 'onChange' which is declared here on type 'IntrinsicAttributes & TextFieldProps'

AmarKaramehmedovic commented:
@antonioorct @mduvnjakag04 I have fixed everything but couldn't manage to add State without getting an error on "onChange" on non date fields: 
Error: The expected type comes from property 'onChange' which is declared here on type 'IntrinsicAttributes & TextFieldProps'

I am not sure if I am doing everything correct with State...

antonioorct commented:
A basic google search turned up with [this](https://github.com/mui/material-ui/issues/19011)   
Have you tried this?

antonioorct commented:
If that doesn't work, you should send us the whole error message. This one is not saying too much. But i can assume it's something with material ui's types

AmarKaramehmedovic commented:
@antonioorct here are full errors:
In the VS Code: 
![image](https://user-images.githubusercontent.com/81163780/184354624-4345158f-8497-4ad2-bf12-509a7f5e1a6e.png)

In the browser:
![image](https://user-images.githubusercontent.com/81163780/184354815-8d347e8f-3d70-4d57-bc04-bc7c038c7a72.png)

antonioorct commented:
The error message clearly shows what's wrong: the function type you are passing as a prop does not match the function type the component is asking for. Change the function you are passing to the prop to match what it is asking for. You can see that either from material ui's API documentation or by ctrl+left click on the `onChange` prop

AmarKaramehmedovic committed:
Fixed errors using state with mui components

Fixed errors using state with mui components. This is the only way it works.

AmarKaramehmedovic commented:
@antonioorct I don't know if you saw commit notification and comment but I have fixed the errors with state and mui components. This is the only way it works. Hardly, but I found some similar solutions (but very deprecated) on GitHub issues.

antonioorct committed:
Add correct types to onChange handlers for input fields

antonioorct commented:
I've fixed your errors regarding the type.

![image](https://user-images.githubusercontent.com/14859262/184537193-3e5836bd-875d-4e00-a12a-4511d6a28393.png)
By passing an anonymous function ineline with the prop and hovering over the argument you can see the type that is inferred by the compiler. So you can just copy and paste that to the `handleChange` function

antonioorct commented:
I think the PR is good to go. You can go ahead and merge
